docs/quickstart.rst

```diff
@@ -0,0 +1,553 @@
+		    'Pig': {'email': 'guinea.pig@example.com', 'name': 'Guinea Pig', 'affiliation': 'Pig University'},
+		    'Bunny': {'email': 'bugs.bunny@example.com', 'name': 'Bugs Bunny', 'affiliation': 'Bunny Inc.'}
+		},
+		'groups': {
+		    'Admins': ['Pig'],
+		    'Everybody': ['Pig', 'Bunny'],
+		}
+	    }
+	}
+
+	MULTIPASS_PROVIDER_MAP = {
+	    'test_auth_provider': 'test_identity_provider'
+	}
+	
+	MULTIPASS_IDENTITY_INFO_KEYS = ['email', 'name', 'affiliation']
+
+We also need to specify the template with a login form for our provider:
+
+.. code-block:: python
+
+	MULTIPASS_LOGIN_FORM_TEMPLATE = 'login_form.html'
+	
+Implementing providers
+~~~~~~~~~~~~~~~~
+
+Let's create our authentication provider class, which should inherit from ``AuthProvider``.
+We should also specify the login form class (which inherits from ``FlaskForm``) which we use for login of this provider.
+
+.. code-block:: python
+
+	class StaticLoginForm(FlaskForm):
+	    username = StringField('Username', [DataRequired()])
+	    password = PasswordField('Password', [DataRequired()])
+
+	class StaticAuthProvider(AuthProvider):
+	    login_form = StaticLoginForm
+    
+.. _local_login:
+
+Local provider login
+~~~~~~~~~~~~~~~~
+
+The process of handling URLs for login is the same as with external providers therefore please check the :ref:`external_login` part of this guide.
+
+The only difference is that  ``multipass.process_login`` calls method ``_login_form`` which renders a template specified in ``MULTIPASS_LOGIN_FORM_TEMPLATE`` with the ``login_form`` specified in the authentication provider class.
+
+Once the form is submitted method ``process_local_login`` of authentication provider class is called. In this method you have to implement your authentication logic. 
+
+You should raise ``MultipassException`` in case of failed validation. If the validation was successful, ``AuthInfo`` object should be created and passed to ``multipass.handle_auth_success``. Bellow is ``process_local_login`` method from our example provider ``StaticAuthProvider``:
+ 
+.. code-block:: python
+
+	def process_local_login(self, data):
+	    username = data['username']
+	    password = self.settings['identities'].get(username)
+	    if password is None:
+		raise AuthenticationFailed('No such user')
+	    if password != data['password']:
+	        raise AuthenticationFailed('Invalid password.')
+	    auth_info = AuthInfo(self, username=data['username'])
+	    return self.multipass.handle_auth_success(auth_info)
+
+
+Identification
+~~~~~~~~~~~~~~~~
+
+The next step after successful authentication is assigning an identity to the user. That's a job of identity provider so let's have a look how to implement one.
+
+Your identity provider should inherit from ``IdentityProvider`` base class. The most important method it has to implement is ``get_identity_from_auth``. Which accepts ``AuthInfo`` object as an argument and returns the corresponding identity (object of ``IdentityInfo``) based on identifier. 
```

Comment:
There should be a comma before "Which", not a full stop.
"based on **an** identifier"